explanation/dnssec/dnssec.md

```diff
@@ -0,0 +1,254 @@
+
+![Stub Resolver](../images/ubuntu-stub-resolver.png)
+
+When an application needs to translate a hostname to an IP address, it uses standard glibc calls for that job. That is called a stub resolver, or simply a "dns client". This is a very simple client in the sense that it does not perform recursive queries: it expects to dispatch the DNS query to a recursive DNS server, which will do all the hard work.
+
+In Ubuntu, the default stub resolver is `systemd-resolved`. That's a daemon, running locally, and listening on port 53/udp on IP 127.0.0.53. The system is configured to use that as its nameserver via `/etc/resolv.conf`:
+
+    nameserver 127.0.0.53
+    options edns0 trust-ad
+
+This stub resolver has its own configuration for which recursive DNS servers to use. That can be seen with the command `resolvectl`. For example:
+
+    Global
+             Protocols: -LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported
+      resolv.conf mode: stub
+
+    Link 12 (eth0)
+        Current Scopes: DNS
+             Protocols: +DefaultRoute -LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported
+    Current DNS Server: 10.10.17.1
+           DNS Servers: 10.10.17.1
+            DNS Domain: lxd
+
+This configuration is usually provided via DHCP, but could also be set via other means. In this particular example, the DNS server that the stub resolver (`systemd-resolved`) will use for all queries that go out on that network interface is 10.10.17.1. The output above also has `DNSSEC=no/unsupported`: we will get back to that in a moment, but it means that `systemd-resolved` is not performing the DNSSEC cryptographic validation.
+
+Given what we have:
+ * an application
+ * stub resolver ("dns client")
+ * recursive DNS server in the local network
+ * several other DNS servers in the internet that will be queried by our recursive DNS server
+
+Where does the DNSSEC validation happen? Who is responsible?
+
+Well, any DNS server can perform the validation. Let's look at two scenarios, and what it means in each case.
+
+### Validating Resolver
+
+When a recursive DNS server is also performing DNSSEC validation, it's called a *Validating Resolver*. That will typically be the DNS server on your local network, at your company, or in some cases at your ISP.
+
+![Validating Resolver](../images/ubuntu-local-validating-resolver.png)
+
+This is the case if you install the `bind9` DNS server: the default configuration is to act as a validating resolver. This can be seen in `/etc/bind/named.conf.options` after installing the `bind9` package:
+
+    options {
+        ...
+        dnssec-validation auto;
+        ...
+    };
+
+The key aspect of this deployment type is that the network between the stub resolver and the validating resolver has to be trusted, otherwise this is not useful. The Validating Resolver will be performing all the DNSSEC checks, and only returning a response if it was validated, but that response could have been altered on that "last mile" network segment.
```

Comment:
"A critical aspect of this deployment model is the trust in the network segment between the stub resolver and the validating resolver. If this network is compromised, the security benefits of DNSSEC can be undermined. While the validating resolver performs DNSSEC checks and returns only verified responses, the response could still be tampered with on the final ("last mile") network segment."

Reply:
just wow